### PR TITLE
Enable XEN backends

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**
This PR includes the following changes:
1) update Xen-tools recipe. Move out proc-xen from the xencommons and make it as a separate package
2) add needed tools to rpi5 image
3) enable backends in linux kernel
4) update u-boot repo and srcrev (just to avoid extra PRs )

### CODE REVIEW

File: recipes-bsp/u-boot/u-boot-tools_2024.04.bb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 require recipes-bsp/u-boot/u-boot.inc
 require recipes-bsp/u-boot/u-boot-common.inc
-
-SRCREV = "13c1100335e40acb1066e074eb061387fd103c36"
+SRC_URI = "git://source.denx.de/u-boot/u-boot.git;protocol=https;branch=rpi5-2024.04-xt"
```
Comment (lorc) [inline recipes-bsp/u-boot/u-boot-tools_2024.04.bb]: I am not sure that DENX repo has rpi-2024.04-xt branch. How you tested this?

**Comment by GrygiriiS** [comment]
@oleksiimoisieiev please, provide test description and prof of working network in guest.
Please, drop u-boot patch from this PR

File: recipes-bsp/u-boot/u-boot-tools_2024.04.bb (hunk 1 above)
Comment (oleksiimoisieiev) [reply recipes-bsp/u-boot/u-boot-tools_2024.04.bb]: dropped this commit and will create a separate PR as was requested by @GrygiriiS

**Comment by oleksiimoisieiev** [comment]
> @oleksiimoisieiev please, provide test description and prof of working network in guest. Please, drop u-boot patch from this PR

Dropped

**Review by lorc — APPROVED** [review]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**